wezterm-gui/src/overlay/copy.rs

```diff
@@ -322,18 +322,43 @@ impl CopyRenderable {
     fn select_to_cursor_pos(&mut self) {
         self.clamp_cursor_to_scrollback();
         if let Some(start) = self.start {
+            // FIXME: how to get the line length from here??
+            // I don't think it's currently possible..
+            // Maybe the SelectionCoordinate should accept an enum with:
+            // - LineColumn::PhysicalStart
+            // - LineColumn::PhysicalEnd
+            // - LineColumn::LogicalStart
+            // - LineColumn::LogicalEnd
+            // (need a better name for 'LineColumn'..)
             let start = SelectionCoordinate {
-                x: start.x,
+                x: match self.selection_mode {
+                    // here we'd use LineColumn::LogicalStart
+                    SelectionMode::Line => 0,
+                    _ => start.x,
+                },
                 y: start.y,
             };
 
             let end = SelectionCoordinate {
-                x: self.cursor.x,
+                x: match self.selection_mode {
+                    // here we'd use LineColumn::LogicalEnd
+                    SelectionMode::Line => 50, // FIXME: remove arbitrary value!
+                    _ => self.cursor.x,
+                },
                 y: self.cursor.y,
             };
+            // debugs to remove..
+            dbg!(self.selection_mode);
+            dbg!(start);
+            dbg!(end);
+
+            // FIXME: when end is above start, the x of start/end should be swapped..
+            // => how to properly do that?
 
             self.adjust_selection(start, SelectionRange { start, end });
         } else {
+            // FIXME: need to handle this as well?
```

Comment:
This is the case where there is no start for the selection.
I don't think there is anything more you need to do for this case